nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -370,12 +370,12 @@ def optimizer_zero_grad(self, *args, **kwargs):
         return
 
     def validation_step(self, dataloader_iter, batch_idx):
-        # Prefetch the dataloader_iter before fwd_bwd func to avoid PP rank 2 from waiting indefinitely when PP rank 1 reaches the end of dataloader_iter
-        dataloader_iter, done = self._prefetch(dataloader_iter)
-        if done:
-            return
         mode = 'test' if self.trainer.testing else 'val'
-        batch = next(dataloader_iter)
+        # Add try except to catch the end of the iterator and exit
+        try:
```

Comment:
what about all other models that inherit from enc_dec? T5 etc, are those handled properly? they dont seem to be changed in this PR

Reply:
@aklife97 wrt `MegatronGPTPromptLearningModel`, it's used only in `DialogueGPTClassificationModel` and `DialogueGPTGenerationModel`. Both of them have their own `validation_step` defined that are using batch (not `dataloader_iter`). So we should be good.

Reply:
I meant generally, not specific to `MegatronGPTPromptLearningModel`
are there other models that might be affected?

Reply:
also, all validation_steps should have this try except for StopIteration right?

Reply:
Yes it's been taken care of either with just `try, except` or with `_val_iterator_done` depending on pre-training or downstream in all megatron models that are using `dataloader_iter`.

Reply:
cool!